### PR TITLE
Origin/vt pair rate 190819

### PR DESCRIPTION
**Pull request by tien7668**

### CODE REVIEW

File: src/js/containers/Exchange/Exchange.js [hunk 1]
```diff
@@ -67,11 +69,11 @@ export default class Exchange extends React.Component {
         this.props.dispatch(exchangeActions.throwErrorSourceAmount(constants.EXCHANGE_CONFIG.sourceErrors.rate, this.props.translate("error.handle_amount")))
         return;
       }
-    } 
+    }
+    var tokens = this.props.tokens
```
Comment (andrewsource147) [inline src/js/containers/Exchange/Exchange.js]: Why moving it out of bracket?

File: src/js/services/limit_order/limit_order.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { timeout, calcInterval, getFormattedDate, getCookie } from "../../utils/common"
 import { LIMIT_ORDER_CONFIG } from "../../services/constants";
 import BLOCKCHAIN_INFO from "../../../../env";
+import * as converters from "../../utils/converter";
+import * as exchangeActions from "../../actions/exchangeActions";
```
Comment (andrewsource147) [inline src/js/services/limit_order/limit_order.js]: If we dont use it please remove

File: src/js/services/limit_order/limit_order.js [hunk 3]
```diff
@@ -422,3 +425,28 @@ export function updateFavoritePairs(base, quote, to_fav){
     })
   })
 }
+
+export function getTokenPrice(symbol){
+    return new Promise((resolve, rejected) => {
+        timeout(MAX_REQUEST_TIMEOUT,
+          fetch("https://api.kyber.network/token_price?currency=ETH", {
```
Comment (andrewsource147) [inline src/js/services/limit_order/limit_order.js]: Please dont hardcord endpoint api.kyber.network

File: src/js/services/limit_order/limit_order.js [hunk 4]
```diff
@@ -422,3 +425,28 @@ export function updateFavoritePairs(base, quote, to_fav){
     })
   })
 }
+
+export function getTokenPrice(symbol){
```
Comment (andrewsource147) [inline src/js/services/limit_order/limit_order.js]: This func should be removed to services/ethereum/nodeProviders/CachedServerProvider.js

File: src/js/services/limit_order/limit_order.js [hunk 5]
```diff
@@ -422,3 +425,28 @@ export function updateFavoritePairs(base, quote, to_fav){
...
+              headers: {
+                  Accept: 'application/json',
+                  'Content-Type': 'application/json',
+              },
+          }))
+          .then(response => response.json())
+          .then((result) => {
+              for (var i = 0; i < result.data.length; i++){
+                  if (result.data[i].symbol == symbol){
+                      resolve(converters.calculateRate(1 ,result.data[i].price ))
```
Comment (andrewsource147) [inline src/js/services/limit_order/limit_order.js]: if you want to convert token to token wei, please use func toTWei

File: src/js/containers/Exchange/Exchange.js (hunk 1 above)
Comment (tien7668) [reply src/js/containers/Exchange/Exchange.js]: moving back into bracket

File: src/js/services/limit_order/limit_order.js (hunk 2 above)
Comment (tien7668) [reply src/js/services/limit_order/limit_order.js]: Removed

File: src/js/services/limit_order/limit_order.js (hunk 4 above)
Comment (tien7668) [reply src/js/services/limit_order/limit_order.js]: Removed

**Pull request merged**